TestNameservers is flaky

Very occasionally (1 in 15 runs so far)

```
--- FAIL: TestNameservers-4 (2.77s)
        Error Trace:    nameserver_test.go:95
			nameserver_test.go:149
			nameserver_test.go:191
	Error:  	Not equal: "51.51.253.7" (expected)
			        != "" (actual)
		
FAIL
coverage: 60.2% of statements
```

On my VM with 4 cores I get one failure in about 1 hour:
```
^Cvagrant@twilkie-test-1:~/src/github.com/weaveworks/weave/nameserver$ while true; do ../test/units.sh -slow; done
ok  	github.com/weaveworks/weave/nameserver	53.243s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	43.687s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	52.608s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	33.929s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	43.315s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	52.691s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	54.456s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	52.590s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	42.775s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.322s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.003s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	54.311s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	41.966s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.998s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	51.021s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	52.524s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	52.954s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	53.669s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	42.693s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.573s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	34.318s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	44.519s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	63.501s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	42.248s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	43.382s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	54.161s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	54.548s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	52.929s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	53.143s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	43.980s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	44.573s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	47.935s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	51.575s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	44.216s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	44.320s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	43.113s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.640s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	43.743s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	52.525s	coverage: 61.4% of statements
--- FAIL: TestNameservers-4 (15.44s)
        Error Trace:    nameserver_test.go:96
			nameserver_test.go:150
			nameserver_test.go:192
	Error:  	Not equal: "103.44.176.76 111.223.1.2" (expected)
			        != "111.223.1.2" (actual)
		
FAIL
coverage: 61.4% of statements
FAIL	github.com/weaveworks/weave/nameserver	20.080s
ok  	github.com/weaveworks/weave/nameserver	62.970s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	34.366s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.650s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	41.682s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	43.978s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	51.332s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	42.106s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.837s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	42.675s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.002s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.466s	coverage: 60.9% of statements
ok  	github.com/weaveworks/weave/nameserver	44.117s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	42.523s	coverage: 61.4% of statements
ok  	github.com/weaveworks/weave/nameserver	53.718s	coverage: 61.4% of statements
```

--- a/nameserver/nameserver_test.go
+++ b/nameserver/nameserver_test.go
@@ -73,7 +73,7 @@ func (m mapping) Addrs() []address.Address {
 func TestNameservers(t *testing.T) {
 	//common.SetLogLevel("debug")
 
-	lookupTimeout := 20 // ms
+	lookupTimeout := 10 // ms
 	nameservers, grouter := makeNetwork(50)
 	defer stopNetwork(nameservers, grouter)
 	nameserversByName := map[router.PeerName]*Nameserver{}
@@ -173,10 +173,6 @@ func TestNameservers(t *testing.T) {
 	}
 
 	for i := 0; i < 1000; i++ {
-		if i%10 == 0 {
-			grouter.Flush()
-		}
-
 		r := rand.Float32()
 		switch {
 		case r < 0.1:
@@ -194,6 +190,8 @@ func TestNameservers(t *testing.T) {
 		case 0.9 <= r:
 			doReverseLookup()
 		}
+
+		grouter.Flush()
 	}
 }
 

